drivers/mock/driver.go

```diff
@@ -152,16 +152,16 @@ type TaskConfig struct {
 	StartErrRecoverable bool `codec:"start_error_recoverable"`
 
 	// StartBlockFor specifies a duration in which to block before returning
-	StartBlockFor time.Duration `codec:"start_block_for"`
+	StartBlockFor string `codec:"start_block_for"`
```

Comment:
What do you think of creating a duration representations of these and a helper for parsing TaskConfig. I started doing this for similar parsing in another driver.

I was thinking something like this:
```
startBlockForDur time.Duration
...
func decodeTaskConfig(cfg *drivers.TaskConfig) (*TaskConfig, err) {
    var tc TaskConfig
    if err := cfg.DecodeDriverConfig(&tc); err != nil {
	    return nil, err
    }

    if tc.StartBlockFor != "" {
        dur, err := time.ParseDuration(tc.StartBlockFor)
        if err != nil {
            return nil, err
        }
        tc.startBlockForDur = dur
    }

    // Other duration vars as well
```

Reply:
That's a useful pattern if other drivers need it, so we can extract handlers.

It seems to me that mock driver is a bit special in that the complexity here stems from tests using duration values (e.g. `2 * time.Second`) as well as string duration (e.g. `"2s"`), while hopefully this is not an option for other drivers.

Unifying approach in mock and other drivers would require either to ensure that we use strings for duration consistently in tests, which is fine but modifies many tests, or risk allowing users to insert nanotime values in config, not ideal.

Reply:
It's a shame we can't model durations in hcl2 so we can validate it statically before reaching the client (or at least before reaching StartTask).

I think this approach is maintaining the validation status quo of pre-0.9, so that's good enough.

Reply:
As I added error handling, I moved to using materialized views of the duration fields and used the pattern @nickethier suggested (with using a helper function).